presto-main/src/main/java/com/facebook/presto/spiller/FileSingleStreamSpiller.java

```diff
@@ -144,11 +145,15 @@ private void writePages(Iterator<Page> pageIterator)
             while (pageIterator.hasNext()) {
                 Page page = pageIterator.next();
                 spilledPagesInMemorySize += page.getSizeInBytes();
-                SerializedPage serializedPage = serde.serialize(page);
-                long pageSize = serializedPage.getSizeInBytes();
-                localSpillContext.updateBytes(pageSize);
-                spillerStats.addToTotalSpilledBytes(pageSize);
-                writeSerializedPage(output, serializedPage);
+                // page serialization requires  page.getSizeInBytes() + Integer.BYTES to fit in an integer
+                splitPage(page, DEFAULT_MAX_PAGE_SIZE_IN_BYTES).stream()
```

Comment:
We should probably add a similar fix to the storage based spiller

CC: @wenleix